src/aws/osml/model_runner/inference/http_detector.py

```diff
@@ -148,19 +148,18 @@ def find_features(self, payload: BufferedReader, metrics: MetricsLogger) -> Feat
                         metrics.put_metric(MetricLabels.RETRIES, retry_count, str(Unit.COUNT.value))
                     return geojson.loads(response.data.decode("utf-8"))
         except RetryError as err:
-            self.error_count += 1
             if isinstance(metrics, MetricsLogger):
                 metrics.put_metric(MetricLabels.ERRORS, 1, str(Unit.COUNT.value))
             logger.error(f"Retry failed - failed due to {err}")
             logger.exception(err)
+            raise err
```

Comment:
Raising an error will gracefully exit the application. Therefore, FeatureCollection will never be returned. I would advise undoing this.

Reply:
The only place this is called is in the tile worker - which will not exit the application as it stands now, it will only add the tile raising the exception to the failed_tiles list. Does that make sense?